### DISCUSSION TITLE
Flickering only when using capture device?

### DISCUSSION THREAD

**Discussion by supagold** [awawa-dev/HyperHDR#274] [Q&A]
Hi, I'm attempting to build a HyperHDR setup using an SK6812 strip, but anytime I plug an HDMI source into my capture device I suddenly get a lot of flickering, especially on the final led strip in my chain. Solid colors and the rainbow animation from HyperHDR run beautifully.

Here's my hardware setup:

- Pi 3 running off of a dedicated AC adapter
- ESP32 running HyperSerialWLED, connected via micro usb cable to pi for power and data. Uses a CP9102 serial chip, and I'm running it through a TI 74AHCT125 for level shifting. I've tied ground from the led strip back into the ground pin on the esp32.
- BTR Lighting SK6812, 280 LEDs in the strip. These are four separate strips soldered with 18awg at the corners of the tv, with power injected at both ends
- I'm using a Mean Well 5V 10A PSU. I measure very close to 5V all across the strip. 
- my capture card is a MS2109 clone. i don't know much about setting this up, but I see my content when I check it in the live preview

I have automatic brightness limiting off in WLED. I've tried changing the baud rate between the Pi and ESP32, and re-soldering the connections into the last strip. (Which is definitely the worst culprit on the flickering, but I can make out some slight flicker in other areas.) Any suggestions on what else I might look at?

**Comment by awawa-dev** [comment]
Hi
Exemplary construction of the set. nothing to complain about. Don't change baudrate: the AWA protocol has a protection so there is no communication errors so no random flickering. Still the main suspect is the grabber/power supply/damaged LED. The grabber has almost none power consumption when it is not capturing: disable it in the video control and try to plug in/out HDMI cable to eliminate some electrical issues. And check out what's happening when you turn on grabber in HyperHDR again with and without HDMI cable. Second possibility: most of effects are using very low refresh rate and they have their own smoothing settings (~25Hz). The grabber on the other hand is using rather high refresh rate and when the smoothing is enabled it produces usually even more frames. If some LEDs in the strips are damaged they can cause such effect increased with the higher refresh rate. Disable smoothing and choose some low capturing rate for USB grabber.

**Comment by supagold** [comment]
- Unchecking Enable USB Capture, then plugging and unplugging the hdmi seems to make no difference. 
- Enabling Enable USB Capture, then plugging and unplugging the hdmi doesn't seem to make a difference, except that sometimes the test pattern is being captured (flickering) and other times I think it detects that no cable is present and drops back to rainbow animation. (no flickering)
- Turning off smoothing seems to introduce subtle flickering even into the rainbow animation. Capture seems stable <30fps on the capture card.
- Setting Latch Time to 22ms seems to remove the flickering. I wasn't sure where to find smoothing at first, so I tried this setting, even tho I'm not sure precisely what the setting does, beyond that it's involved in signal speed.
- I tried fully replacing the final section of the LED strip that was exhibiting the most flickering, with seemingly no effect. 
 
Any ideas what this might imply? (Thank you for the help, btw!)

**Reply by supagold** [reply]
Hi, I had to take a break working on this, but I'd like to get back to it, and was hoping you might have some pointers for me to try. I tried a new level shifter, since the one i had was from an old project, and a little ragged. No change as far as I can tell. I've got another sk6812 strip coming tomorrow, on the chance that this strip has a bad LED or solder connection. If you have any ideas based on my last post in the thread, I'd really appreciate any guidance you could share. Thank you!

**Reply by awawa-dev** [reply]
No, I don't have any more ideas. You have all the required hardware and it should work unless something is broken, badly connected (bad wiring especially around 74AHCT125 or common ground, too long cables etc) or Rpi has insufficient power but I can't help remotely with it.
